# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/components/sessionstore/test/browser.ini
# Commit: d29c1fad7862
# Full Hash: d29c1fad7862d61ef7c66d5989caa5ed0af4f011
# Author: Mike Conley <mconley@mozilla.com>
# Date: 2023-03-03 21:37:37
# Regressor Bug: 1423125
# File Overlap Count: 1
# Description:
#   Bug 1423125 - Restore previous session if undoCloseTab is called with no more tabs to restore. r=dao
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171377
# ==============================================================================

diff -r c81bb857ccff -r d29c1fad7862 browser/components/sessionstore/test/browser.ini
--- a/browser/components/sessionstore/test/browser.ini	Fri Mar 03 14:06:07 2023 +0000
+++ b/browser/components/sessionstore/test/browser.ini	Fri Mar 03 14:06:43 2023 +0000
@@ -72,121 +72,10 @@
 #disabled-for-intermittent-failures--bug-766044, browser_459906_sample.html
 #disabled-for-intermittent-failures--bug-765389, browser_461743_sample.html
 
-[browser_aboutPrivateBrowsing.js]
-[browser_aboutSessionRestore.js]
-skip-if =
-  verify && debug && os == "win"
-  verify && debug && os == "mac"
-[browser_async_duplicate_tab.js]
-support-files = file_async_duplicate_tab.html
-[browser_async_flushes.js]
-support-files = file_async_flushes.html
-run-if = crashreporter
-[browser_async_remove_tab.js]
-skip-if = !sessionHistoryInParent
-[browser_attributes.js]
-[browser_backup_recovery.js]
-https_first_disabled = true
-skip-if =
-  verify && debug && os == "linux"
-[browser_broadcast.js]
-https_first_disabled = true
-[browser_capabilities.js]
-[browser_cleaner.js]
-[browser_closedId.js]
-[browser_crashedTabs.js]
-https_first_disabled = true
-skip-if =
-  !crashreporter
-  verify
-  win10_2004 # high frequency intermittent, Bug 1684120 - timed out
-  os == "mac" # high frequency intermittent
-[browser_firefoxView_restore.js]
-[browser_formdata_max_size.js]
-[browser_multiple_select_after_load.js]
-[browser_pinned_tabs.js]
-skip-if =
-  debug
-  ccov # Bug 1625525
-[browser_restore_srcdoc.js]
-[browser_restore_tabless_window.js]
-[browser_restored_window_features.js]
-[browser_unrestored_crashedTabs.js]
-skip-if =
-  !crashreporter
-[browser_revive_crashed_bg_tabs.js]
-https_first_disabled = true
-skip-if =
-  !crashreporter
-[browser_dying_cache.js]
-skip-if = (os == "win") # bug 1331853
-[browser_dynamic_frames.js]
-[browser_firefoxView_selected_restore.js]
-[browser_formdata.js]
-skip-if =
-  verify && debug
-[browser_formdata_cc.js]
-[browser_formdata_format.js]
-skip-if = !debug && (os == "linux") # Bug 1535645
-[browser_formdata_password.js]
-support-files = file_formdata_password.html
-[browser_formdata_xpath.js]
-[browser_frametree.js]
-https_first_disabled = true
-[browser_frame_history.js]
-[browser_global_store.js]
-[browser_history_persist.js]
-[browser_label_and_icon.js]
-https_first_disabled = true
-skip-if =
-  apple_silicon # Disabled due to bleedover with other tests when run in regular suites; passes in "failures" jobs
-  apple_catalina && !debug # Bug 1638958
-  win10_2004 && bits == 64 && !debug # Bug 1638958
-  os == "linux" && !debug # Bug 1638958
-  os == "win" && os_version == "6.1" # Skip on Azure - frequent failure
-[browser_merge_closed_tabs.js]
-[browser_old_favicon.js]
-https_first_disabled = true
-[browser_page_title.js]
-[browser_pending_tabs.js]
-[browser_privatetabs.js]
-[browser_purge_shistory.js]
-skip-if = !sessionHistoryInParent # Bug 1271024
-[browser_replace_load.js]
-skip-if = true # Bug 1646894
-[browser_restore_redirect.js]
-https_first_disabled = true
-[browser_restoreTabContainer.js]
-[browser_restore_cookies_noOriginAttributes.js]
-[browser_scrollPositions.js]
-https_first_disabled = true
-skip-if =
-  !fission
-  os == "linux" # Bug 1716445
-[browser_scrollPositionsReaderMode.js]
-[browser_sessionHistory.js]
-https_first_disabled = true
-support-files =
-  file_sessionHistory_hashchange.html
-skip-if =
-  os == "linux" # Bug 1775608
-  os == "mac" && os_version == "10.15" && debug # Bug 1775608
-[browser_sessionStorage.js]
-skip-if =
-  os == "win" && os_version == "6.1" # Skip on Azure - frequent failure
-[browser_sessionStorage_size.js]
-[browser_sizemodeBeforeMinimized.js]
-[browser_tab_label_during_restore.js]
-https_first_disabled = true
-[browser_swapDocShells.js]
-[browser_switch_remoteness.js]
-[browser_upgrade_backup.js]
-skip-if =
-  debug
-  asan
-  tsan
-  verify && debug && os == "mac" # Bug 1435394 disabled on Linux, OSX and Windows
-[browser_windowRestore_perwindowpb.js]
+[browser_1234021.js]
+[browser_1284886_suspend_tab.js]
+[browser_1446343-windowsize.js]
+skip-if = os == "linux" # Bug 1600180
 [browser_248970_b_perwindowpb.js]
 # Disabled because of leaks.
 # Re-enabling and rewriting this test is tracked in bug 936919.
@@ -213,6 +102,11 @@
 [browser_463205.js]
 [browser_463206.js]
 [browser_464199.js]
+# Disabled for frequent intermittent failures
+[browser_464620_a.js]
+skip-if = true
+[browser_464620_b.js]
+skip-if = true
 [browser_465215.js]
 [browser_465223.js]
 [browser_466937.js]
@@ -243,7 +137,6 @@
 [browser_579879.js]
 skip-if = (os == "linux" && (debug||asan)) # Bug 1234404
 [browser_581937.js]
-[browser_586147.js]
 [browser_586068-apptabs.js]
 [browser_586068-apptabs_ondemand.js]
 skip-if = (verify && (os == "mac" || os == "win"))
@@ -255,6 +148,7 @@
 [browser_586068-select.js]
 [browser_586068-window_state.js]
 [browser_586068-window_state_override.js]
+[browser_586147.js]
 [browser_588426.js]
 [browser_590268.js]
 [browser_590563.js]
@@ -279,6 +173,11 @@
 [browser_618151.js]
 [browser_623779.js]
 [browser_624727.js]
+[browser_625016.js]
+skip-if =
+  os == "mac" # Disabled on OS X:
+  os == "linux" # linux, Bug 1348583
+  os == "win" && debug # Bug 1430977
 [browser_628270.js]
 [browser_635418.js]
 [browser_636279.js]
@@ -304,83 +203,82 @@
 [browser_739805.js]
 [browser_819510_perwindowpb.js]
 skip-if = true # Bug 1284312, Bug 1341980, bug 1381451
-[browser_not_collect_when_idle.js]
-
-# Disabled for frequent intermittent failures
-[browser_464620_a.js]
-skip-if = true
-[browser_464620_b.js]
-skip-if = true
-
-[browser_625016.js]
-skip-if =
-  os == "mac" # Disabled on OS X:
-  os == "linux" # linux, Bug 1348583
-  os == "win" && debug # Bug 1430977
-
 [browser_906076_lazy_tabs.js]
 https_first_disabled = true
 skip-if = os == "linux" && os_version == "18.04" # bug 1446464
 [browser_911547.js]
-[browser_1284886_suspend_tab.js]
+[browser_aboutPrivateBrowsing.js]
+[browser_aboutSessionRestore.js]
+skip-if =
+  verify && debug && os == "win"
+  verify && debug && os == "mac"
+[browser_async_duplicate_tab.js]
+support-files = file_async_duplicate_tab.html
+[browser_async_flushes.js]
+support-files = file_async_flushes.html
+run-if = crashreporter
+[browser_async_remove_tab.js]
+skip-if = !sessionHistoryInParent
 [browser_async_window_flushing.js]
 https_first_disabled = true
 skip-if =
   os == "linux" # Bug 1775616
   os == "mac"  # Bug 1775616
   os == "win" && !debug  # Bug 1775616
-[browser_focus_after_restore.js]
-[browser_forget_async_closings.js]
-https_first_disabled = true
-[browser_movePendingTabToNewWindow.js]
-https_first_disabled = true
-[browser_multiple_navigateAndRestore.js]
-skip-if = os == "linux" && debug #Bug 1570468
-[browser_newtab_userTypedValue.js]
-skip-if = verify && debug
-[browser_parentProcessRestoreHash.js]
-https_first_disabled = true
-tags = openUILinkIn
-skip-if =
-  os == "win" && os_version == "6.1" # Skip on Azure - frequent failure
-[browser_send_async_message_oom.js]
-skip-if = sessionHistoryInParent # Tests that the frame script OOMs, which is unused when SHIP is enabled.
-[browser_sessionStoreContainer.js]
-[browser_windowStateContainer.js]
-[browser_1234021.js]
-[browser_remoteness_flip_on_restore.js]
+[browser_attributes.js]
 [browser_background_tab_crash.js]
 https_first_disabled = true
 run-if = crashreporter
 # Disabled on debug for frequent intermittent failures:
-[browser_undoCloseById.js]
-skip-if = debug
-[browser_docshell_uuid_consistency.js]
-
+[browser_backup_recovery.js]
+https_first_disabled = true
+skip-if =
+  verify && debug && os == "linux"
+[browser_bfcache_telemetry.js]
+[browser_broadcast.js]
+https_first_disabled = true
+[browser_capabilities.js]
+[browser_cleaner.js]
+[browser_closedId.js]
 [browser_closed_objects_changed_notifications_tabs.js]
 [browser_closed_objects_changed_notifications_windows.js]
-[browser_duplicate_history.js]
-[browser_tabicon_after_bg_tab_crash.js]
-skip-if =
-  !crashreporter
-[browser_tabs_in_urlbar.js]
-https_first_disabled = true
-
 [browser_cookies.js]
 [browser_cookies_legacy.js]
 [browser_cookies_privacy.js]
-[browser_speculative_connect.js]
-[browser_1446343-windowsize.js]
-skip-if = os == "linux" # Bug 1600180
-[browser_restore_reversed_z_order.js]
-skip-if = true #Bug 1455602
 [browser_cookies_sameSite.js]
-
-[browser_urlbarSearchMode.js]
-[browser_restore_container_tabs_oa.js]
-[browser_restore_private_tab_os.js]
-[browser_reopen_all_windows.js]
+[browser_crashedTabs.js]
 https_first_disabled = true
+skip-if =
+  !crashreporter
+  verify
+  win10_2004 # high frequency intermittent, Bug 1684120 - timed out
+  os == "mac" # high frequency intermittent
+[browser_docshell_uuid_consistency.js]
+[browser_duplicate_history.js]
+[browser_duplicate_tab_in_new_window.js]
+[browser_dying_cache.js]
+skip-if = (os == "win") # bug 1331853
+[browser_dynamic_frames.js]
+[browser_firefoxView_restore.js]
+[browser_firefoxView_selected_restore.js]
+[browser_focus_after_restore.js]
+[browser_forget_async_closings.js]
+https_first_disabled = true
+[browser_formdata.js]
+skip-if =
+  verify && debug
+[browser_formdata_cc.js]
+[browser_formdata_format.js]
+skip-if = !debug && (os == "linux") # Bug 1535645
+[browser_formdata_max_size.js]
+[browser_formdata_password.js]
+support-files = file_formdata_password.html
+[browser_formdata_xpath.js]
+[browser_frame_history.js]
+[browser_frametree.js]
+https_first_disabled = true
+[browser_global_store.js]
+[browser_history_persist.js]
 [browser_ignore_updates_crashed_tabs.js]
 https_first_disabled = true
 run-if = crashreporter
@@ -388,7 +286,104 @@
   asan
   os == "win" && fission && verify # bug 1709907
   os == "mac" && fission # Bug 1711008; high frequency intermittent
-[browser_bfcache_telemetry.js]
+[browser_label_and_icon.js]
+https_first_disabled = true
+skip-if =
+  apple_silicon # Disabled due to bleedover with other tests when run in regular suites; passes in "failures" jobs
+  apple_catalina && !debug # Bug 1638958
+  win10_2004 && bits == 64 && !debug # Bug 1638958
+  os == "linux" && !debug # Bug 1638958
+  os == "win" && os_version == "6.1" # Skip on Azure - frequent failure
+[browser_merge_closed_tabs.js]
+[browser_movePendingTabToNewWindow.js]
+https_first_disabled = true
+[browser_multiple_navigateAndRestore.js]
+skip-if = os == "linux" && debug #Bug 1570468
+[browser_multiple_select_after_load.js]
+[browser_newtab_userTypedValue.js]
+skip-if = verify && debug
+[browser_not_collect_when_idle.js]
+[browser_old_favicon.js]
+https_first_disabled = true
+[browser_page_title.js]
+[browser_parentProcessRestoreHash.js]
+https_first_disabled = true
+tags = openUILinkIn
+skip-if =
+  os == "win" && os_version == "6.1" # Skip on Azure - frequent failure
+[browser_pending_tabs.js]
+[browser_pinned_tabs.js]
+skip-if =
+  debug
+  ccov # Bug 1625525
+[browser_privatetabs.js]
+[browser_purge_shistory.js]
+skip-if = !sessionHistoryInParent # Bug 1271024
+[browser_remoteness_flip_on_restore.js]
+[browser_reopen_all_windows.js]
+https_first_disabled = true
+[browser_replace_load.js]
+skip-if = true # Bug 1646894
+[browser_restoreTabContainer.js]
+[browser_restore_container_tabs_oa.js]
+[browser_restore_cookies_noOriginAttributes.js]
+[browser_restore_pageProxyState.js]
+[browser_restore_private_tab_os.js]
+[browser_restore_redirect.js]
+https_first_disabled = true
+[browser_restore_reversed_z_order.js]
+skip-if = true #Bug 1455602
+[browser_restore_session_in_undoCloseTab.js]
+[browser_restore_srcdoc.js]
+[browser_restore_tabless_window.js]
+[browser_restored_window_features.js]
+[browser_revive_crashed_bg_tabs.js]
+https_first_disabled = true
+skip-if =
+  !crashreporter
+[browser_scrollPositions.js]
+https_first_disabled = true
+skip-if =
+  !fission
+  os == "linux" # Bug 1716445
+[browser_scrollPositionsReaderMode.js]
+[browser_send_async_message_oom.js]
+skip-if = sessionHistoryInParent # Tests that the frame script OOMs, which is unused when SHIP is enabled.
+[browser_sessionHistory.js]
+https_first_disabled = true
+support-files =
+  file_sessionHistory_hashchange.html
+skip-if =
+  os == "linux" # Bug 1775608
+  os == "mac" && os_version == "10.15" && debug # Bug 1775608
+[browser_sessionStorage.js]
+skip-if =
+  os == "win" && os_version == "6.1" # Skip on Azure - frequent failure
+[browser_sessionStorage_size.js]
+[browser_sessionStoreContainer.js]
+[browser_sizemodeBeforeMinimized.js]
+[browser_speculative_connect.js]
+[browser_swapDocShells.js]
+[browser_switch_remoteness.js]
+[browser_tab_label_during_restore.js]
+https_first_disabled = true
+[browser_tabicon_after_bg_tab_crash.js]
+skip-if =
+  !crashreporter
+[browser_tabs_in_urlbar.js]
+https_first_disabled = true
+[browser_undoCloseById.js]
+skip-if = debug
+[browser_unrestored_crashedTabs.js]
+skip-if =
+  !crashreporter
+[browser_upgrade_backup.js]
+skip-if =
+  debug
+  asan
+  tsan
+  verify && debug && os == "mac" # Bug 1435394 disabled on Linux, OSX and Windows
+[browser_urlbarSearchMode.js]
 [browser_userTyped_restored_after_discard.js]
-[browser_restore_pageProxyState.js]
-[browser_duplicate_tab_in_new_window.js]
+[browser_windowRestore_perwindowpb.js]
+[browser_windowStateContainer.js]